common/src/main/java/com/microsoft/identity/common/AndroidPlatformComponents.java

```diff
@@ -242,6 +246,64 @@ public IMultiTypeNameValueStorage getFileStore(final @NonNull String storeName)
         return SharedPreferencesFileManager.getSharedPreferences(mContext, storeName, null);
     }
 
+    @Override
+    public INameValueStorage<String> getMultiProcessStringStore(final @NonNull String storeName) {
+        final SharedPreferences sharedPreferences = mContext.getSharedPreferences(storeName, Context.MODE_MULTI_PROCESS);
+        return new SharedPrefStringNameValueStorage(new IMultiTypeNameValueStorage() {
+            @Override
+            public void putString(String key, String value) {
+                sharedPreferences.edit().putString(key, value);
+            }
+
+            @Override
+            public String getString(String key) {
+                return sharedPreferences.getString(key, null);
+            }
+
+            @Override
+            public void putLong(String key, long value) {
+                sharedPreferences.edit().putString(key, Long.toString(value));
+            }
+
+            @Override
+            public long getLong(String key) {
+                try {
+                    if (!sharedPreferences.contains(key)) {
+                        return 0;
+                    }
+                    return Long.parseLong(sharedPreferences.getString(key, "0"));
+                } catch (final NumberFormatException nfe) {
+                    return 0;
+                }
+            }
+
+            @Override
+            public Map<String, String> getAll() {
+                return (Map<String, String>) sharedPreferences.getAll();
+            }
+
+            @Override
+            public Iterator<Map.Entry<String, String>> getAllFilteredByKey(Predicate<String> keyFilter) {
+                return null;
+            }
+
+            @Override
+            public boolean contains(String key) {
+                return sharedPreferences.contains(key);
+            }
+
+            @Override
+            public void clear() {
+                sharedPreferences.edit().clear().commit();
+            }
+
+            @Override
+            public void remove(String key) {
+                sharedPreferences.edit().remove(key).commit();
+            }
+        });
```

Comment:
I believe we already have implementations of SharedPreferences for INameValueStorage, can't we just use those here? (just with a different operating mode)

Reply:
Let me see if I can find it.

Reply:
The SharedPreferencesFileManager is a bit of a Swiss-army-knife right now.  The main thing it does, I have to defeat - I absolutely must create new instances of this every time I read, or we risk not having cross-process visibility.